### DISCUSSION TITLE
Is it possible to use systemd socket activation with librespot-java?

### DISCUSSION THREAD

**Discussion by GeorgeHJ** [librespot-org/librespot-java#433] [Q&A]
Currently I have a `systemd` user unit for `librespot-java` which pretty much just runs a wrapper shell script.

My question was, whether there's a particular way to have `systemd` listen for something like the official Spotify client or `spotify-tui` so that the back-end is only running when there's a controller for it?

**Comment by zugazagoitia** [comment]
AFAIK you can do it through systemd if they are also a Unit, using `Wants` or `Requires`. You could also code that logic into the wrapper script.
